File: PlacementCellBackend/Services/CRUD/Interfaces/ICompanyEmployeeService.cs
using PlacementCellBackend.DTOs.CompanyEmployee;

namespace PlacementCellBackend.Services.CRUD.Interfaces;

public interface ICompanyEmployeeService
{
    Task<IEnumerable<CompanyEmployeeDto>> GetAllCompanyEmployeesAsync();
    Task<CompanyEmployeeDto?> GetCompanyEmployeeByIdAsync(string id);
    Task<bool> CreateCompanyEmployeeAsync(CompanyEmployeeCreateDto companyEmployee);
    Task<bool> UpdateCompanyEmployeeAsync(string id, CompanyEmployeeCreateDto companyEmployee);
    Task<bool> DeleteCompanyEmployeeAsync(string id);
    bool CompanyEmployeeExists(string id);
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/CRUD/Interfaces/ICompanyEmployeeService.java
package com.example.placementicsbackend.services.CRUD.Interfaces;

import com.example.placementicsbackend.dtos.CompanyEmployee.CompanyEmployeeCreateDto;
import com.example.placementicsbackend.dtos.CompanyEmployee.CompanyEmployeeDto;
import java.util.List;

public interface ICompanyEmployeeService {

    List<CompanyEmployeeDto> getAllCompanyEmployees();

    CompanyEmployeeDto getCompanyEmployeeById(String id);

    boolean createCompanyEmployee(CompanyEmployeeCreateDto companyEmployee);

    boolean updateCompanyEmployee(String id, CompanyEmployeeCreateDto companyEmployee);

    boolean deleteCompanyEmployee(String id);

    boolean companyEmployeeExists(String id);
}
